x-pack/plugins/observability/server/services/slo/create_slo.ts

```diff
@@ -10,31 +10,41 @@ import uuid from 'uuid';
 import { SLO } from '../../types/models';
 import { ResourceInstaller } from './resource_installer';
 import { SLORepository } from './slo_repository';
-import { TransformInstaller } from './transform_installer';
-
+import { TransformManager } from './transform_manager';
 import { CreateSLOParams, CreateSLOResponse } from '../../types/schema';
 
 export class CreateSLO {
   constructor(
     private resourceInstaller: ResourceInstaller,
     private repository: SLORepository,
-    private transformInstaller: TransformInstaller,
-    private spaceId: string
+    private transformManager: TransformManager
   ) {}
 
   public async execute(sloParams: CreateSLOParams): Promise<CreateSLOResponse> {
     const slo = this.toSLO(sloParams);
 
-    await this.resourceInstaller.ensureCommonResourcesInstalled(this.spaceId);
+    await this.resourceInstaller.ensureCommonResourcesInstalled();
     await this.repository.save(slo);
 
+    let sloTransformId;
     try {
-      await this.transformInstaller.installAndStartTransform(slo, this.spaceId);
+      sloTransformId = await this.transformManager.install(slo);
     } catch (err) {
       await this.repository.deleteById(slo.id);
       throw err;
     }
 
+    try {
+      await this.transformManager.start(sloTransformId);
+    } catch (err) {
+      await Promise.all([
+        this.transformManager.uninstall(sloTransformId),
+        this.repository.deleteById(slo.id),
+      ]);
```

Comment:
💬 If an error happens while starting the transform, we remove the transform and remove the SLO from the repository

Reply:
We could add a transient error (timeout, connection error) retry mechanism 👍🏻 If it's fine with you, I've created an issue to track this and do in another PR: https://github.com/elastic/kibana/issues/141082

For the other type of error (like invalid transformer, bad index, ...), I don't think we can do much.